### PR TITLE
Fix Engine shutdown races

### LINKED ISSUE #438: SybilProtection accesses closed storage
Similar to #435

```
panic: trying to access closed kvstore: failed to retrieve from KV store: failed to get value for epoch 1: failed to load committee for epoch 1

goroutine 2544 [running]:
github.com/iotaledger/iota-core/pkg/protocol/sybilprotection/sybilprotectionv1/performance.(*Tracker).LoadCommitteeForEpoch(0x722a18fbfda604?, 0xd13c839a?)
	/scratch/pkg/protocol/sybilprotection/sybilprotectionv1/performance/performance.go:117 +0xae
github.com/iotaledger/iota-core/pkg/protocol/sybilprotection/sybilprotectionv1/performance.(*Tracker).isCommitteeMember(0xc000b9e5b0, 0x365e, {0x3f, 0x74, 0x20, 0x68, 0x56, 0xfe, 0x50, 0xbc, ...})
	/scratch/pkg/protocol/sybilprotection/sybilprotectionv1/performance/performance.go:241 +0x67
github.com/iotaledger/iota-core/pkg/protocol/sybilprotection/sybilprotectionv1/performance.(*Tracker).TrackValidationBlock(0xc000b9e5b0, 0xc00f433b80)
	/scratch/pkg/protocol/sybilprotection/sybilprotectionv1/performance/performance.go:71 +0x1d2
github.com/iotaledger/iota-core/pkg/protocol/sybilprotection/sybilprotectionv1.(*SybilProtection).TrackValidationBlock(0xc0337fd4a0?, 0xc032f24120?)
	/scratch/pkg/protocol/sybilprotection/sybilprotectionv1/sybilprotection.go:112 +0x17
github.com/iotaledger/iota-core/pkg/protocol/engine.(*Engine).acceptanceHandler.func1(0xc033767ed8?)
	/scratch/pkg/protocol/engine/engine.go:403 +0x5e
github.com/iotaledger/hive.go/runtime/event.(*Event1[...]).Trigger.func1.1()
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231010133617-cdbd5387e2af/event/events.go:86 +0x1f
github.com/iotaledger/hive.go/runtime/workerpool.(*Task).run(0xc020c51140)
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231010133617-cdbd5387e2af/workerpool/task.go:48 +0x72
github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).workerReadLoop(0xc000b9e460)
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231010133617-cdbd5387e2af/workerpool/workerpool.go:190 +0x25
github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).worker(0xc000b9e460)
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231010133617-cdbd5387e2af/workerpool/workerpool.go:170 +0x50
created by github.com/iotaledger/hive.go/runtime/workerpool.(*WorkerPool).startWorkers in goroutine 139
	/go/pkg/mod/github.com/iotaledger/hive.go/runtime@v0.0.0-20231010133617-cdbd5387e2af/workerpool/workerpool.go:162 +0x25

```

### PR DESCRIPTION
**Pull request by karimodm**
Fixes #438 and #435

### CODE REVIEW

**Review by hmoog — APPROVED** [review]
LGTM

File: pkg/protocol/engine/blockdag/inmemoryblockdag/blockdag.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func (b *BlockDAG) setupBlock(block *blocks.Block) {
 	block.ForEachParent(func(parent iotago.Parent) {
 		parentBlock, exists := b.blockCache.Block(parent.ID)
 		if !exists {
-			panic("cannot setup block without existing parent")
```
Comment (jonastheis) [inline pkg/protocol/engine/blockdag/inmemoryblockdag/blockdag.go:81]: Would be better to log (error handler for now) an error here no?

File: pkg/protocol/engine/engine.go [hunk 2]
```diff
@@ -236,8 +237,14 @@ func (e *Engine) Shutdown() {
 		e.CommitmentFilter.Shutdown()
 		e.Scheduler.Shutdown()
 		e.Retainer.Shutdown()
-		e.Storage.Shutdown()
+		go func() {
+			time.Sleep(5 * time.Second)
```
Comment (jonastheis) [inline pkg/protocol/engine/engine.go]: Remove debug

**Review by jonastheis — CHANGES_REQUESTED** [review]

**Review by alexsporn — APPROVED** [review]
Fix the comments from Jonas, rest LGTM

**Review by jonastheis — APPROVED** [review]

**Pull request merged**